src/CSharp/CodeCracker/Design/MakeMethodStaticAnalyzer.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Linq;
+using System.Collections.Immutable;
+
+namespace CodeCracker.CSharp.Design
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class MakeMethodStaticAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string Title = "Use static method";
+        internal const string MessageFormat = "Make '{0}' method static.";
+        internal const string Category = SupportedCategories.Design;
+        const string Description = "If you are not referencing any instance variable your method may become static. That will require";
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId.MakeMethodStatic.ToDiagnosticId(),
+            Title,
+            MessageFormat,
+            Category,
+            DiagnosticSeverity.Warning,
+            true,
+            description: Description,
+            helpLinkUri: HelpLink.ForDiagnostic(DiagnosticId.MakeMethodStatic));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) =>
+            context.RegisterSyntaxNodeAction(LanguageVersion.CSharp6, AnalyzeMethod, SyntaxKind.MethodDeclaration);
+
+        private void AnalyzeMethod(SyntaxNodeAnalysisContext context)
+        {
+            if (context.IsGenerated()) return;
+            var method = (MethodDeclarationSyntax)context.Node;
+            if (method.Modifiers.Any(SyntaxKind.StaticKeyword)) return;
```

Comment:
The analyzer should also immediately return if there are any of this method modifiers:
* `virtual`
* `override`
* `new`